solidity/test/maintainer/Maintainer.test.ts

```diff
@@ -0,0 +1,3577 @@
+                    })
+
+                    after(async () => {
+                      await restoreSnapshot()
+                    })
+
+                    it("should succeed", async () => {
+                      await expect(outcome).to.not.be.reverted
+                    })
+
+                    it("should refund ETH", async () => {
+                      const resolvedOutcome = await outcome
+
+                      const postThirdPartyBalance = await provider.getBalance(
+                        thirdParty.address
+                      )
+
+                      const diff = postThirdPartyBalance.sub(
+                        resolvedOutcome.initThirdPartyBalance
+                      )
+                      expect(diff).to.be.gt(0)
+                      expect(diff).to.be.lt(
+                        ethers.utils.parseUnits("9000000", "gwei") // 0,009 ETH
+                      )
+                    })
+                  })
+                }
+              )
+            })
+          })
+        })
+      })
+    })
+  })
+
+  describe("notifyWalletCloseable", () => {
+    before(async () => {
+      await createSnapshot()
+    })
+
+    after(async () => {
+      walletRegistry.requestNewWallet.reset()
+
+      await restoreSnapshot()
+    })
+    context("when called by an unauthorized third party", async () => {
+      it("should revert", async () => {
+        await expect(
+          maintainerProxy
+            .connect(thirdParty)
+            .notifyWalletCloseable(
+              ecdsaWalletTestData.pubKeyHash160,
+              NO_MAIN_UTXO
+            )
+        ).to.be.revertedWith("Caller is not authorized")
+      })
+    })
+
+    context("when called by an authorized party", async () => {
+      context("when the reported wallet is not the active one", () => {
+        context("when wallet is in Live state", () => {
+          before(async () => {
+            await createSnapshot()
+
+            await maintainerProxy
+              .connect(governance)
+              .authorize(walletRegistry.address)
+            await reimbursementPool
+              .connect(governance)
+              .authorize(maintainerProxy.address)
+
+            // Set the deposit dust threshold to 0.0001 BTC, i.e. 100x smaller than
+            // the initial value in the Bridge in order to save test Bitcoins.
+            await bridge.setDepositDustThreshold(10000)
+
+            await bridge.setWallet(ecdsaWalletTestData.pubKeyHash160, {
+              ecdsaWalletID: ecdsaWalletTestData.walletID,
+              mainUtxoHash: ethers.constants.HashZero,
+              pendingRedemptionsValue: 0,
+              createdAt: await lastBlockTime(),
+              movingFundsRequestedAt: 0,
+              closingStartedAt: 0,
+              pendingMovedFundsSweepRequestsCount: 0,
+              state: walletState.Live,
+              movingFundsTargetWalletsCommitmentHash: ethers.constants.HashZero,
+            })
```

Comment:
Do we have a test where this state is not an expected initial one? Maybe we could even move it to the main test setup function?

Reply:
We have 6 tests that use `ecdsaWalletTestData.pubKeyHash160`, but their states are different depending on the scenario: `walletState.Live`, `walletState.MovingFunds` & `walletState.Unknown`. Also, these test setups are spread across different test functions and were ported in the same form from the already existing tests. I feel like in this case we're not gaining much by extracting it. Unless you have a strong opinion on that, I'd leave it as is.